### PR TITLE
Sneak peek: Recipe app

### PR DESCRIPTION
**Pull request by noelfoka**
## Sneak peek: Recipe app

In this branch, we implemented:

- [ ] Login page and registration page using devise.
- [ ] Recipes list:
- [ ] Public recipe list
- [ ] Recipe details
- [ ] Make sure there are no N+1 queries happening.
- [ ] Create a navigation menu that allows users to open all of the pages you created.
- [ ] Write unit and integration tests
- [ ] Food list
- [ ] General shopping list view

### CODE REVIEW

**Comment by topeogunleye** [comment]
Hello @noelfoka, @Kccendana, please can you fix your error on line 3 of recipe_food_spec.rb. it is too long

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# class UsersController < ApplicationController
+#   def index; end
+
+#   def destroy
+#     @user.destroy
+#     respond_to do |format|
+#       format.html { redirect_to root_url, notice: 'User was successfully destroyed.' }
+#     end
+#   end
+# end
```
Comment (MahmoudBakr23) [inline app/controllers/users_controller.rb]: - [ ] Please remove any unused code for a cleaner and more optimized code base. And if you're not using the whole file of code, I recommend deleting it. 🙏

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @noelfoka and @Kccendana 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Well done on creating the controllers' restful methods and n+1 query problems
- Great job with authentication so far
- Nice work with the models' validations
- Good work with the database setup, migrations, and the routes
- Well-written unit and integration tests
- Descriptive PR
- Correct Linters

However, there is always room for improvement.

> I'd like to remind you that your long hours of work are appreciated. However, the code review process is for improving your code, not criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

N/A

Cheers, and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AbelG101 — APPROVED** [review]
# Hi @noelfoka & @Kccendana  👋, 

**Your project is complete!  There is nothing else to say other than... it's time to merge it. 
Congratulations! 🎉 🥳🥇**

## Highlights
- Commented codes have been removed ✔
- App functions properly ✔
- Correct use of Git Flow ✔
- All linter checks are passing ✔
- Good PR summary ✔

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**